lib/gooddata/models/project_creator.rb

```diff
@@ -86,6 +107,59 @@ def migrate_datasets(spec, opts = {})
           replaced_maqls + (ca_maql ? [ca_maql] : [])
         end
 
+        def migrate_datasets_async(spec, opts = {})
+          opts = { client: GoodData.connection }.merge(opts)
+          dry_run = opts[:dry_run]
+          replacements = opts['maql_replacements'] || opts[:maql_replacements] || {}
+
+          _, project = GoodData.get_client_and_project(opts)
+
+          bp = ProjectBlueprint.new(spec)
+          response = project.maql_diff(blueprint: bp, params: [:includeGrain])
+
+          GoodData.logger.debug("projectModelDiff") { response.pretty_inspect }
+          chunks = response['projectModelDiff']['updateScripts']
+          return [] if chunks.empty?
+
+          ca_maql = response['projectModelDiff']['computedAttributesScript'] if response['projectModelDiff']['computedAttributesScript']
+          ca_chunks = ca_maql && ca_maql['maqlDdlChunks']
+
+          maqls = pick_correct_chunks(chunks, opts)
+          replaced_maqls = apply_replacements_on_maql(maqls, replacements)
+
+          to_poll = []
+          unless dry_run
+            errors = []
+            replaced_maqls.each do |replaced_maql_chunks|
+              begin
+                replaced_maql_chunks['updateScript']['maqlDdlChunks'].each do |chunk|
+                  GoodData.logger.debug(chunk)
+                  to_poll << project.execute_maql_async(chunk)
+                end
+              rescue => e
+                puts "Error occured when executing MAQL, project: \"#{project.title}\" reason: \"#{e.message}\", chunks: #{replaced_maql_chunks.inspect}"
+                errors << e
+                next
+              end
+            end
+
+            if ca_chunks && opts[:execute_ca_scripts]
+              begin
```

Comment:
Identical code found in project_creator.rb:94, project_creator.rb:147 (mass = 136)